### PR TITLE
Adding args during initialization in model trainer

### PR DESCRIPTION
**Pull request by alaydshah**
I propose this modification after realizing that _**args**_ parameter will be needed for most of the functions when we are trying advanced techniques for e.g, **freezing backbone**, as shown in the snippets below:

```
    def get_model_params(self):
        if self.args.backbone_freezed:
            logging.info('Initializing model; Backbone Freezed')
            return self.model.encoder_decoder.cpu().state_dict()
        else:
            logging.info('Initializing end-to-end model')
            return self.model.cpu().state_dict()
```

```
    def set_model_params(self, model_parameters):
        if self.args.backbone_freezed:
            logging.info('Updating Global model; Backbone Freezed')
            self.model.encoder_decoder.load_state_dict(model_parameters)
        else:
            logging.info('Updating Global model')
            self.model.load_state_dict(model_parameters)
```

Currently for the segmentation part, I am just passing the args once during initialization and using it in all the functions. And for now, I thought to make it optional at all places so none of the dependent files are affected. Do let me know if there is a better way to do it.

### CODE REVIEW

File: fedml_core/trainer/model_trainer.py [hunk 1]
```diff
@@ -8,9 +8,11 @@ class ModelTrainer(ABC):
        2. This class can be used in both server and client side
        3. This class is an operator which does not cache any states inside.
     """
-    def __init__(self, model):
+    def __init__(self, model, args=None):
+        # TODO: Make args mandatory during initialization
         self.model = model
         self.id = 0
+        self.args = args
```
Comment (mitashimpi) [inline fedml_core/trainer/model_trainer.py:15]: Once we set the args as a member variable why do we still need to pass it in the class methods?

File: fedml_core/trainer/model_trainer.py [hunk 2]
```diff
@@ -24,9 +26,11 @@ def set_model_params(self, model_parameters):
         pass
 
     @abstractmethod
-    def train(self, train_data, device, args):
+    def train(self, train_data, device, args=None):
```
Comment (mitashimpi) [inline fedml_core/trainer/model_trainer.py:29]: Once we set the args as a member variable why do we still need to pass it in the class methods?

Comment (alaydshah) [reply fedml_core/trainer/model_trainer.py:29]: Setting it as a member variable is something I am proposing to do. Before this commit, it was just passed as argument to two specific functions. However, there are other projects using these core trainer files and if I remove args right away then their code will break. So, we need to remove it after they have made the changes as well. Have added _TODO comments_ mentioning the same. Also, I have mentioned the same reasoning in bit more detailed manner in the PR conversation itself incase you missed it.

Comment (mitashimpi) [reply fedml_core/trainer/model_trainer.py:29]: Yeah, I saw that. Got it

**Review by mitashimpi — APPROVED** [review]

**Review by keerti-1997 — APPROVED** [review]

**Pull request merged**